ctags.lua

```diff
@@ -129,11 +132,53 @@ local function get_match(word, path)
 	end
 end
 
-local function open_file(path, line)
-	vis:command(string.format('open %s', path))
+local function goto_tag(path, line)
+	local pos = {
+		path = vis.win.file.path,
+		line = vis.win.selection.line,
+		col  = vis.win.selection.col,
+	}
+
+	-- check path twice instead of testing vim:command() as it returns true even
+	-- if edit fails to open the file
+	if path ~= vis.win.file.path then
+		vis:command(string.format('e %s', path))
+	end
+	if path ~= vis.win.file.path then
+		return
+	end
 	vis.win.selection:to(tonumber(line), 1)
+
+	npos = npos + 1
+	positions[npos] = pos
 end
 
+local function pop_pos()
+	if npos < 1 then
+		return
+	end
+
+	local path = positions[npos].path
+	local line = positions[npos].line
+	local col  = positions[npos].col
+
+	-- check path twice instead of testing vim:command() as it returns true even
+	-- if edit fails to open the file
+	if path ~= vis.win.file.path then
+		vis:command(string.format('e %s', path))
+	end
+	if path ~= vis.win.file.path then
+		return
+	end
+	vis.win.selection:to(line, col)
```

Comment:
The code to open a file and set the cursor in the `goto_tag` and `pop_pos` functions looks very similar and should be moved to a separate function (something like `goto_pos`?).

Reply:
Yep, good point.